src/main/java/uk/gov/hmcts/reform/sscs/service/hearings/UpdateHearingCaseUpdater.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.hmcts.reform.sscs.service.hearings;
+
+import feign.FeignException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sscs.ccd.client.CcdClient;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.service.SscsCcdConvertService;
+import uk.gov.hmcts.reform.sscs.ccd.service.UpdateCcdCaseService;
+import uk.gov.hmcts.reform.sscs.exception.ListingException;
+import uk.gov.hmcts.reform.sscs.exception.UpdateCaseException;
+import uk.gov.hmcts.reform.sscs.helper.mapping.OverridesMapping;
+import uk.gov.hmcts.reform.sscs.helper.service.HearingsServiceHelper;
+import uk.gov.hmcts.reform.sscs.idam.IdamService;
+import uk.gov.hmcts.reform.sscs.model.HearingEvent;
+import uk.gov.hmcts.reform.sscs.model.HearingWrapper;
+import uk.gov.hmcts.reform.sscs.model.hearings.HearingRequest;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingRequestPayload;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HmcUpdateResponse;
+import uk.gov.hmcts.reform.sscs.service.HmcHearingApiService;
+import uk.gov.hmcts.reform.sscs.service.holder.ReferenceDataServiceHolder;
+
+import static java.util.Objects.isNull;
+import static uk.gov.hmcts.reform.sscs.helper.mapping.HearingsMapping.buildHearingPayload;
+import static uk.gov.hmcts.reform.sscs.helper.service.HearingsServiceHelper.getHearingId;
+
+@Component
+@Slf4j
+public class UpdateHearingCaseUpdater extends HearingSaveActionBase {
+
+    private final HmcHearingApiService hmcHearingApiService;
+
+    private final IdamService idamService;
+
+    public UpdateHearingCaseUpdater(CcdClient ccdClient,
+                                    SscsCcdConvertService sscsCcdConvertService, HmcHearingApiService hmcHearingApiService,
+                                    ReferenceDataServiceHolder refData,
+                                    IdamService idamService) {
+        super(ccdClient, sscsCcdConvertService, refData);
+        this.hmcHearingApiService = hmcHearingApiService;
+        this.idamService = idamService;
+    }
+
+    public void updateHearingAndCase(HearingRequest hearingRequest) throws ListingException, UpdateCaseException {
+        HearingEvent event = HearingsServiceHelper.getHearingEvent(hearingRequest.getHearingState());
+        try {
+            updateCase(
+                Long.valueOf(hearingRequest.getCcdCaseId()),
+                event.getEventType().getCcdType(),
+                idamService.getIdamTokens(),
+                hearingRequest
+            );
+        } catch (FeignException ex) {
+            UpdateCaseException exception = new UpdateCaseException(
```

Comment:
test coverage?

Reply:
This is covered in the following test

UpdateHearingCaseUpdaterTest -> updateHearingAndCase

Reply:
Added tests.